ee/server/service/mdm.go

```diff
@@ -66,6 +67,15 @@ func getAppleBMAccountDetail(ctx context.Context, depStorage storage.AllStorage,
 	depClient := apple_mdm.NewDEPClient(depStorage, ds, logger)
 	res, err := depClient.AccountDetail(ctx, apple_mdm.DEPName)
 	if err != nil {
+		var authErr *depclient.AuthError
+		if errors.As(err, &authErr) {
+			// authentication failure means that the configured Apple BM certificate
+			// and/or token are invalid. Fail with a 400 Bad Request.
+			return nil, ctxerr.Wrap(ctx, &fleet.BadRequestError{
+				Message:     "The Apple Business Manager certificate or server token is invalid. Restart Fleet with a valid certificate and token. See https://fleetdm.com/docs/using-fleet/mdm-setup#apple-business-manager-abm for help.",
```

Comment:
For now, the frontend will not display this user-friendly message, but at least it will be ready to do so when https://github.com/fleetdm/fleet/issues/11932 gets implemented.